site/content/blog/2024-07-17-script-scan-rules/index.md

```diff
@@ -0,0 +1,222 @@
+The alert raised by each script is automatically populated with the details from the metadata.
+However, you can still override metadata for particular alerts if you want.
+For example,
+
+```js
+function scan(helper, msg, src) {
+  // ... 
+  foundKeys = ["my-secret-key", "my-other-secret-key"]
+  helper
+    .newAlert()
+    .setEvidence(foundKeys[0])
+    .setOtherInfo(`Other instances: ${foundKeys.slice(1).toString()}`)
+    .setMessage(msg)
+    .raise();
+}
+```
+
+While the code examples in this blog post are written in JS, ZAP scripts written in any supported programming language (e.g. Python, Groovy, Kotlin, etc.) may be exposed as scan rules.
+
+## Community Scripts
+
+All the passive scripts and most of the active scripts in the [community-scripts](https://github.com/zaproxy/community-scripts) repository (and add-on) have been updated to implement the `getMetadata()` function.
+
+The [alerts page](/docs/alerts/) on this website (which is auto-generated from scan rules) now contains information about the alerts raised by the community scripts too.
+
+It should be noted that all these scripts will now need to use the GraalVM engine (provided by the `graaljs` add-on), as template literals are not supported by Nashorn.
+
+If you have any scripts written for Nashorn, we recommend you switch to using the GraalVM engine.
+You should be able to switch your engine for the majority of your scripts without having to modify them.
+
+If you do not want to use GraalVM, you can still expose your scripts as scan rules by constructing the [ScanRuleMetadata](https://github.com/zaproxy/zap-extensions/blob/839d5c7df7432c1f4748275daacd1d4e8812a51a/addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/scanrules/ScanRuleMetadata.java) object manually with its setters.
+
+## Rapid Prototyping
+
+Last month, we published a new passive scan rule to detect if a script in your webapp was being loaded from the malicious Polyfill.io domain.
+You can read the full blog post about it [here](/blog/2024-06-27-polyfill.io-script-detection/).
+
+The scan rule was [written in Java](https://github.com/zaproxy/zap-extensions/blob/839d5c7df7432c1f4748275daacd1d4e8812a51a/addOns/pscanrulesBeta/src/main/java/org/zaproxy/zap/extension/pscanrulesBeta/PolyfillCdnScriptScanRule.java) and included in the `pscanrulesBeta` add-on.
+
+Here's the same scan rule written as a script, included here as an example.
+
+```js
+const ScanRuleMetadata = Java.type(
+  "org.zaproxy.addon.commonlib.scanrules.ScanRuleMetadata",
+);
+const Alert = Java.type("org.parosproxy.paros.core.scanner.Alert");
+
+const PLUGIN_ID = 10115;
+
+function getMetadata() {
+  return ScanRuleMetadata.fromYaml(`
+id: ${PLUGIN_ID}
+name: Script Served From Malicious Domain (polyfill)
+description: |
+  The page includes one or more script files loaded from one of the 'polyfill' domains.
+  These are not associated with the polyfill.js library and are known to serve malicious content.
```

Comment:
We could use i18n'd messages here, but I added the full description since this is an example.